e2e/run-test.sh

```diff
@@ -53,28 +53,33 @@ function run_test {
 }
 
 function verify_image {
-  info "Verifying that the image built by Forge has the expected files at the expected paths"
-  local test_dir="/tmp/$namespace/verify_image"
+  local image_name=$1
+  local src_dir=$2
+  local expected_dir=$3
+
+  info "Verifying that the $image_name image built by Forge has the expected files"
+  local registry="localhost:32002"
+  local registry_user=marge
+  local registry_password=simpson
 
   info "Logging in to Docker"
-  echo "simpson" | docker login localhost:32002 -u=marge --password-stdin
+  echo "$registry_password" | docker login $registry -u=$registry_user --password-stdin
 
   info "Copying files from image"
-  mkdir -p "$test_dir/actual"
-  docker cp $(docker create localhost:32002/variable-base-app:latest):/app/app.py "$test_dir/actual/app.py"
-
-  info "Extracting files that are expected to be in the image"
-  mkdir -p "$test_dir/expected"
-  tar -xf "$BASE_DIR/internal/archive/testdata/simple-app.tar" -C "$test_dir/expected"
+  local actual_dir=$(mktemp -d -t actual-XXXXXXXXXX)
+  docker cp $(docker create $registry/$image_name):$src_dir "$actual_dir"
 
-  info "Comparing files from the image with expected files"
-  if ! diff "$test_dir/expected/app.py" "$test_dir/actual/app.py"; then
+  info "Comparing files from the $image_name image with expected files"
+  if ! diff -r "$expected_dir" "$actual_dir"; then
     error  "diff failed"
-    ls -lahR "$test_dir"
+    echo "EXPECTED:"
+    ls -lahR "$expected_dir"
+    echo "ACTUAL:"
+    ls -lahR "$actual_dir"
     exit 1
   fi
 
-  info "Test succeeded"
```

Comment:
Don't we want to keep this line albeit after the following one so we can easily distinguish when an e2e test passes? I think you and I would have a sense that `Files in ...` means the test passes but new devs probably would not, right?

Reply:
good idea